testutils/src/main/java/com/amplifyframework/testutils/SynchronousApi.java

```diff
@@ -0,0 +1,386 @@
+/*
+ * Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amplifyframework.testutils;
+
+import androidx.annotation.NonNull;
+
+import com.amplifyframework.api.ApiCategory;
+import com.amplifyframework.api.graphql.GraphQLRequest;
+import com.amplifyframework.api.graphql.GraphQLResponse;
+import com.amplifyframework.api.graphql.MutationType;
+import com.amplifyframework.api.graphql.SubscriptionType;
+import com.amplifyframework.api.rest.RestOptions;
+import com.amplifyframework.api.rest.RestResponse;
+import com.amplifyframework.core.Amplify;
+import com.amplifyframework.core.ResultListener;
+import com.amplifyframework.core.StreamListener;
+import com.amplifyframework.core.async.Cancelable;
+import com.amplifyframework.core.model.Model;
+import com.amplifyframework.core.model.query.predicate.QueryPredicate;
+import com.amplifyframework.util.Immutable;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * A utility to perform synchronous calls to the {@link ApiCategory}.
+ * This code is not well suited for production use, but is useful in test
+ * code, where we want to make a series of sequential assertions after
+ * performing various operations.
+ */
+public final class SynchronousApi {
+
+    @SuppressWarnings("checkstyle:all") private SynchronousApi() {}
+
+    /**
+     * Gets an instance of the Synchronous API utility.
+     * @return Synchronous API instance
+     */
+    @NonNull
+    public static SynchronousApi instance() {
+        return new SynchronousApi();
```

Comment:
Good point! I had thought about making all of these methods `static` methods, actually. The only reason I didn't saw that I wanted to be able to write (in short hand):
```
SynchronousApi api = SynchronousApi.instance();
api.whatever(...) // "api" is short to write
```
Instead of:
```
SynchronousApi.whatever(...) // SynchronousApi needs to be written each time
```

But, you're right - I could make it a singleton, which would be the ~best of both worlds. Will update.